AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Internal/URLComponents+sigv4Encoding.swift

```diff
@@ -22,15 +22,29 @@ extension URLComponents {
         var sigV4UnreservedCharacters = CharacterSet(charactersIn: "A" ... "Z")
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "a" ... "z"))
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "0" ... "9"))
-        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~.%=&"))
+        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~."))
         return sigV4UnreservedCharacters
     }()
 
-    mutating func percentEncodeQueryBySigV4Rules() {
-        guard let percentEncodedQuery = self.percentEncodedQuery else {
+    private func encodeQueryParamValueBySigV4Rules(_ value: String) -> String {
+        /// removingPercentEncoding returns `nil` if called on a value
+        /// that hasn't been prior encoded
+        let unencoded = value.removingPercentEncoding ?? value
+
+        /// apply sigv4 encoding
+        return unencoded.addingPercentEncoding(
+            withAllowedCharacters: Self.sigV4UnreservedCharacters) ?? value
+    }
+
+    mutating func queryItemsEncodedPerSigV4Rules(_ queryItems: [String: String]?) {
```

Comment:
nit, since this method doesn't return a value, but rather mutates in place, consider renaming to something that is imperative rather than declarative, such as `encodeQueryItemsPerSigV4Rules` or similar. (Similar constructions in the Swift stdlib with things like `sort()` and `sorted()`)